wrappers/javascript/nodejs/src/indyVdr.ts

```diff
@@ -0,0 +1,563 @@
+    const { revocationRegistryId, toTs, fromTs, submitterDid } = serializeArguments(options) as SerializedOptions<
+      typeof options
+    >
+
+    const convertedFromTs = fromTs ?? -1
+
+    handleError(
+      nativeIndyVdr.indy_vdr_build_get_revoc_reg_delta_request(
+        submitterDid,
+        revocationRegistryId,
+        convertedFromTs,
+        toTs,
+        requestHandle
+      )
+    )
+
+    return requestHandle.deref() as number
+  }
+
+  public buildRevocRegDefRequest(options: RevocationRegistryDefinitionRequestOptions): number {
+    const requestHandle = allocateHandleBuffer()
+    const { submitterDid, revocationRegistryDefinitionV1: revocationRegistryDefinition } = serializeArguments(
+      options
+    ) as SerializedOptions<typeof options>
+
+    handleError(
+      nativeIndyVdr.indy_vdr_build_revoc_reg_def_request(submitterDid, revocationRegistryDefinition, requestHandle)
+    )
+
+    return requestHandle.deref() as number
+  }
+
+  public buildCustomRequest(options: CustomRequestOptions): number {
+    const requestHandle = allocateHandleBuffer()
+    const { customRequest } = serializeArguments(options) as SerializedOptions<typeof options>
+
+    handleError(nativeIndyVdr.indy_vdr_build_custom_request(customRequest, requestHandle))
+
+    return requestHandle.deref() as number
+  }
+
+  public buildDisableAllTxnAuthorAgreementsRequest(
+    options: DisableAllTransactionAuthorAgreementsRequestOptions
+  ): number {
+    const requestHandle = allocateHandleBuffer()
+    const { submitterDid } = serializeArguments(options) as SerializedOptions<typeof options>
+
+    handleError(nativeIndyVdr.indy_vdr_build_disable_all_txn_author_agreements_request(submitterDid, requestHandle))
+
+    return requestHandle.deref() as number
+  }
+
+  public buildGetNymRequest(options: GetNymRequestOptions): number {
+    const requestHandle = allocateHandleBuffer()
+    const { dest, submitterDid } = serializeArguments(options) as SerializedOptions<typeof options>
+
+    handleError(nativeIndyVdr.indy_vdr_build_get_nym_request(submitterDid, dest, requestHandle))
+
+    return requestHandle.deref() as number
+  }
+
+  public buildGetSchemaRequest(options: GetSchemaRequestOptions): number {
+    const requestHandle = allocateHandleBuffer()
+    const { schemaId, submitterDid } = serializeArguments(options) as SerializedOptions<typeof options>
+
+    handleError(nativeIndyVdr.indy_vdr_build_get_schema_request(submitterDid, schemaId, requestHandle))
+
+    return requestHandle.deref() as number
+  }
+
+  public buildGetTxnAuthorAgreementRequest(options: GetTransactionAuthorAgreementRequestOptions): number {
+    const requestHandle = allocateHandleBuffer()
+    const { data, submitterDid } = serializeArguments(options) as SerializedOptions<typeof options>
+
+    handleError(nativeIndyVdr.indy_vdr_build_get_txn_author_agreement_request(submitterDid, data, requestHandle))
+
+    return requestHandle.deref() as number
```

Comment:
Is the return type of `deref` always number? In that case could we add this to the method?

Reply:
No it dereferences a buffer, pointer in NodeJS, to its original type